extensions/control-plane/store/cosmos/asset-index-cosmos/src/main/java/org/eclipse/edc/connector/store/azure/cosmos/assetindex/CosmosAssetIndex.java

```diff
@@ -100,19 +103,27 @@ public Asset findById(String assetId) {
     }
 
     @Override
-    public void accept(AssetEntry item) {
+    public StoreResult<Void> accept(AssetEntry item) {
         var assetDocument = new AssetDocument(item.getAsset(), partitionKey, item.getDataAddress());
-        assetDb.createItem(assetDocument);
+        try {
+            assetDb.createItem(assetDocument);
+        } catch (ConflictException ex) {
+            var id = item.getAsset().getId();
+            monitor.debug(() -> String.format("Asset with id %s not found", id));
```

Comment:
Also here just reuse `ASSET_EXISTS_TEMPLATE` and also use the static import for  `format`